### PR TITLE
Batched the ES request for product IDs

### PR DESCRIPTION
**Pull request by lyle-adeptmind**

### CODE REVIEW

File: summary/__main__.py [hunk 1]
```diff
@@ -125,11 +125,17 @@ def fetch_and_process_events(**kwargs):
...
 
+    es_results = []
     fetch_ids_query = {
         "query": {"ids": {"values": product_ids}},
     }
-    es_results = products_es_client.fetch(fetch_ids_query)
-    populate_events_with_products(es_results, events)
+    query_results = products_es_client.fetch(fetch_ids_query)
+    while len(query_results) > 0:
+        es_results.extend(query_results)
```
Comment (SiweiYang) [inline summary/__main__.py:134]: peter buhr?

Comment (lyle-adeptmind) [reply summary/__main__.py:134]: Is this a reference to something I should know about?

File: summary/__main__.py [hunk 2]
```diff
@@ -125,11 +125,17 @@ def fetch_and_process_events(**kwargs):
...
     product_ids = list(chain.from_iterable(map(lambda event: event.get_product_ids(), events)))
 
+    es_results = []
     fetch_ids_query = {
         "query": {"ids": {"values": product_ids}},
     }
-    es_results = products_es_client.fetch(fetch_ids_query)
-    populate_events_with_products(es_results, events)
+    query_results = products_es_client.fetch(fetch_ids_query)
+    while len(query_results) > 0:
```
Comment (jinghe) [inline summary/__main__.py:133]: maybe better to encapsulate the logic inside the client, like `fetch_all`?

Comment (lyle-adeptmind) [reply summary/__main__.py:133]: That's true.  Actually this is on the roadmap as part of a larger refactor I was discussing with Siwei yesterday.  So yes and also not right now.

**Pull request merged**